cmd/mungedocs/links.go

```diff
@@ -29,6 +29,8 @@ import (
 var (
 	// Finds markdown links of the form [foo](bar "alt-text").
 	linkRE = regexp.MustCompile(`\[([^]]*)\]\(([^)]*)\)`)
+	// Finds markdown link typos of the form (foo)[bar]
+	badLinkRE = regexp.MustCompile(`\(([^]()]*)\)\[([^]()]*)\]`)
```

Comment:
I feel like this belongs in util.go and should be more like `mlLinkRE`. And we should annotate the mungeLine with `badLink` or something. Then just check for `badLink` here.  I'm ok doing it here, but as I recall regexps with submatches were measurably slower and should be avoided when they weren't needed. (`mlLinkRE` hits every line, `linkRE` should only hit lines that already passed `mlLinkRE`, so I think that should be the regexp we copy here)